File: src-tauri/src/commands/keep_awake.rs
use super::errors::CommandError;
use crate::AppState;
use tauri::State;

#[tauri::command]
pub fn keep_awake_enable(state: State<AppState>) -> Result<(), CommandError> {
    let mut guard = state
        .keep_awake
        .lock()
        .map_err(|_| CommandError::MutexPoisoned("Keep awake"))?;

    if guard.is_none() {
        let awake = keepawake::Builder::default()
            .display(true)
            .idle(true)
            .reason("HomeKaraoke video playing")
            .app_name("HomeKaraoke")
            .app_reverse_domain("app.homekaraoke")
            .create()
            .map_err(|e| CommandError::External(format!("Failed to enable keep awake: {}", e)))?;

        *guard = Some(awake);
        log::info!("Keep awake enabled");
    }

    Ok(())
}

#[tauri::command]
pub fn keep_awake_disable(state: State<AppState>) -> Result<(), CommandError> {
    let mut guard = state
        .keep_awake
        .lock()
        .map_err(|_| CommandError::MutexPoisoned("Keep awake"))?;

    if guard.is_some() {
        *guard = None;
        log::info!("Keep awake disabled");
    }

    Ok(())
}

File: src/services/keepAwake.ts
import { invoke } from "@tauri-apps/api/core";

export const keepAwakeService = {
  async enable(): Promise<void> {
    try {
      await invoke("keep_awake_enable");
      console.log("[KeepAwake] Enabled via native API");
    } catch (err) {
      console.error("[KeepAwake] Failed to enable:", err);
    }
  },

  async disable(): Promise<void> {
    try {
      await invoke("keep_awake_disable");
      console.log("[KeepAwake] Disabled via native API");
    } catch (err) {
      console.error("[KeepAwake] Failed to disable:", err);
    }
  },
};
